pkg/asset/installconfig/openstack/validation/machinepool.go

```diff
@@ -119,8 +119,13 @@ func validateFlavor(flavor Flavor, req flavorRequirements, fldPath *field.Path)
 	if flavor.VCPUs < req.VCPUs {
 		errs = append(errs, fmt.Sprintf("Must have minimum of %d VCPUs, had %d", req.VCPUs, flavor.VCPUs))
 	}
-	if flavor.Disk < req.Disk {
-		errs = append(errs, fmt.Sprintf("Must have minimum of %d GB Disk, had %d GB", req.Disk, flavor.Disk))
+	// Here we tolerate the minimum storage requirements to come from
+	// Disk or Ephemeral.
+	// The root disk is an ephemeral disk that the base image is copied into
+	// whereas an ephemeral disk offers to the machine a local disk storage
+	// linked to the lifecycle of a VM instance.
+	if flavor.Disk < req.Storage && flavor.Ephemeral < req.Storage {
+		errs = append(errs, fmt.Sprintf("Must have minimum of %d GB Disk, had %d GB for root disk and %d GB for ephemeral", req.Storage, flavor.Disk, flavor.Ephemeral))
```

Comment:
Maybe rework the error message. right now it doesn't read well.  also is it "and" GB for ephemereal or "or"?. The code says "or" (the inverse of &&)

Reply:
I think the message is fine:

```
Must have minimum of 25 GB Disk, had 0 GB for root disk and 15 GB for ephemeral
```
In case of not meeting the requirements, we present what the flavor offers. How would you reword it?

Reply:
@adduarte I improved the message, please let me know if it's ok now. Thanks

Reply:
I can remove that comment, since I didn't touch that code at the end.